### PR TITLE
Fix month validation

### PR DESCRIPTION
**Pull request by bkjohnson**
## Chromatic

https://month-date-validation-fix--60f9b557105290003b387cd5.chromatic.com

## Description
Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1173

This updates the date validation so that if the day is invalid and we don't have a month value, we override the invalid day message with the invalid month message.

The valid day range was being calculated based on the month, since for some months 31 will be valid and for others it won't.

## Testing done

- Storybook :eyes: 

## Screenshots

The memorable date component on storybook has its day filled out but not its month. There is an error message saying "Please enter a month between 1 and 12".

## Acceptance criteria
- [ ]

### CODE REVIEW

File: packages/web-components/src/utils/date-utils.ts [hunk 1]
```diff
@@ -171,15 +175,22 @@ export function validate(
...
   if (!monthYearOnly && (day < minMonths || day > daysForSelectedMonth)) {
     component.invalidDay = true;
-    component.error = `Please enter a day between ${minMonths} and ${daysForSelectedMonth}`;
+    component.error = dayError;
+
+    // We need a valid month in order to determine the day
+    if (!month || month < minMonths || month > maxMonths) {
+      component.invalidMonth = true;
+      component.error = monthError;
+    }
```
Comment (bkjohnson) [inline packages/web-components/src/utils/date-utils.ts:184]: Having this extra check inside the day check is the main change here. The other changes are refactoring to use shared consts.

File: packages/web-components/src/utils/date-utils.ts [hunk 2]
```diff
@@ -177,7 +177,8 @@ export function validate(
     component.invalidDay = false;
   }
 
-  if (month && (month < minMonths || month > maxMonths)) {
+  if ((month && (month < minMonths || month > maxMonths)) ||
+      (!month && component.invalidDay)) {
```
Comment (bkjohnson) [inline packages/web-components/src/utils/date-utils.ts:181]: Override the invalid day message if we don't have a month.

Comment (k80bowman) [reply packages/web-components/src/utils/date-utils.ts:181]: Could you please add some tests around the logic in this file? Right now it looks like we only have a test for the leap year logic.

File: packages/web-components/src/utils/date-utils.ts [hunk 3]
```diff
@@ -177,7 +177,8 @@ export function validate(
     component.invalidDay = false;
   }
 
-  if (month && (month < minMonths || month > maxMonths)) {
+  if ((month && (month < minMonths || month > maxMonths)) ||
```
Comment (jamigibbs) [inline packages/web-components/src/utils/date-utils.ts:180]: Is it possible to add some clarifying comments like the other if statements have in this function?

File: packages/web-components/src/utils/date-utils.ts (hunk 2 above)
Comment (bkjohnson) [reply packages/web-components/src/utils/date-utils.ts:184]: Tests have been added.

File: packages/web-components/src/utils/date-utils.ts (hunk 3 above)
Comment (bkjohnson) [reply packages/web-components/src/utils/date-utils.ts:183]: I added some comments which should help - let me know what you think.

**Review by k80bowman — APPROVED** [review]

**Review by jamigibbs — APPROVED** [review]

**Pull request merged**